### DISCUSSION TITLE
can embedded client be shared by multiple goroutines?

### DISCUSSION THREAD

**Discussion by agufagit** [olric-data/olric#222] [Q&A] [answered]
can the same embedded client be shared by multiple goroutines?  or I have to create a new embedded client for each goroutine?

```go
e := db.NewEmbeddedClient()
```

**Comment by buraksezer** [comment, marked as answer]
Hello @agufagit, both client implementations and DMap operations are thread-safe. An initialized client should be shared between goroutines.
